sync/class.jetpack-sync-actions.php

```diff
@@ -137,6 +146,43 @@ static function do_full_sync( $modules = null ) {
 		self::do_send_pending_data(); // try to send at least some of the data
 	}
 
+	static function minute_cron_schedule( $schedules ) {
+		if( ! isset( $schedules["1min"] ) ) {
+			$schedules["1min"] = array(
+				'interval' => 60,
+				'display' => __( 'Every minute' ) );
+		}
+		return $schedules;
+	}
+
+	// try to send actions for up to a minute
+	static function do_cron_sync() {
+		if ( ! self::sync_allowed() ) {
+			return;
+		}
+
+		self::initialize_sender();
+
+		$start_time = microtime( true );
+		
+		do {
+			$next_sync_time = self::$sender->next_sync_time();
+			
+			// sleep for up to 10s
+			if ( $next_sync_time ) {
+				$delay = $next_sync_time - time() + 1;
+				if ( 0 < $delay && $delay < 10 ) {
```

Comment:
because I don't like the idea of sleeping for more than 10s inside a cron job, as we'd be holding up other tasks. I'm open to making this longer, but I do want to have a maximum amount of time :)